### PR TITLE
#164069314 Comment on article

### PR DESCRIPTION
**Pull request by nyabongoedgar**
**What does this PR do?**
* Adds a feature that enables an authenticated user to make a comment on an article *

**Description of the tasks to be completed?**
* User can comment on an article once logged in. 

**How should this be manually tested?**
* Signup at api/users/ 
* verify account and login at api/users/login/ 
* Create a category, POST "/api/categories/",
                                sample_data={
                                    "category": {
                                        "name": "Religion"}}
                                    

* create an article, POST at  '/api/articles/' 
sample_article = {
            "article": {
                "title": "believer",
                "description": "This test was created on womens day of 2019.",
                "body": "I like to move it move it, I like to move it",
            }
        }
* create the comment(s) by posting data to POST /api/articles/<str:slug>/comments/ and fetch by a get
sample_data_for_comments = {
  "comment": {
    "body": "His name was my name too."
  }
}
Note: Slug is auto created and it is returned when an article is created.
* Delete Comment at
DELETE /api/articles/:slug/comments/:id .  
*
* Update Comment at
PUT /api/articles/:slug/comments/:id .  

**What are the relevant pivot tracker stores?**
* #164069314

### CODE REVIEW

**Review by NLSanyu — COMMENTED** [review]
@nyabongoedgar could you please edit your PR title to fit the Andela conventions?

File: tests/articles/test_comments.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            "article": {
+                "title": "believer",
+                "description": "This test was created on womens day of 2019.",
+                "body": "I like to move it move it, I like to move it",
+            }
+        }
+        self.new_article = {
+            "article": {
+                "title": "believer",
+                "description": "This test was created on womens day of 2019.",
```
Comment (Stanley-Okwii) [inline tests/articles/test_comments.py]: @nyabongoedgar, This is a repetition of code. You can re-use much more from the Base_test. You are writing the same instances which are in the class that you inherit.

File: tests/articles/test_comments.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        comment_reponse = self.client.post(
+            "/api/articles/{}/comments/".format(str(slug)),
+            self.comment_data, format="json")
+
+        get_response = self.client.get(
+            "/api/articles/{}/comments/".format(str(slug)))
+        self.assertEqual(comment_reponse.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(get_response.status_code, status.HTTP_200_OK)
+        self.assertIn("body", get_response.data["Comment"][0])
+        # for item in get_response.data:
```
Comment (Stanley-Okwii) [inline tests/articles/test_comments.py]: @nyabongoedgar , remove this unnecessary comments from here.

File: tests/articles/test_comments.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        get_one_article = self.client.get(
+            "/api/articles/{}/comments/{}/".format(str(slug), comment_id))
+        self.assertEqual(get_one_article.data['id'], comment_id)
+        self.assertEqual(get_one_article.status_code, status.HTTP_200_OK)
+
+    def test_get_mulitple_comments_one_one_slug(self):
+        article_response = self.create_an_article(
+            self.article, self.registration_data)
+        slug = article_response.data["slug"]
+        self.client.post(
```
Comment (Stanley-Okwii) [inline tests/articles/test_comments.py:32]: @nyabongoedgar , create a method in the base_test to help you create comments. Your tests are being cluttered with posting comments.

**Review by Stanley-Okwii — DISMISSED** [review]

**Comment by walimike** [comment]
@nyabongoedgar  Please add screenshots on your PR

**Review by NLSanyu — APPROVED** [review]

**Review by akram256 — APPROVED** [review]
Nice work @nyabongoedgar

**Review by Stanley-Okwii — APPROVED** [review]
Good for merging

**Review by walimike — APPROVED** [review]

**Review by PatrickCmd — APPROVED** [review]

**Pull request merged**